### PR TITLE
fix(client): remove algolia and hot keys modules from landing pages

### PR DESCRIPTION
**Pull request by ahmaxed**
Checklist:

- [ ] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [ ] My pull request targets the `main` branch of freeCodeCamp.
- [ ] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: api-server/src/common/models/user.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ import {
   renderSignInEmail
 } from '../utils';
 
-import { blocklistedUsernames } from '../../server/utils/constants.js';
+import { blocklistedUsernames } from '../../../../config/constants';
```
Comment (raisedadead) [inline api-server/src/common/models/user.js:29]: Is this change intentional?

Comment (ahmadabdolsaheb) [reply api-server/src/common/models/user.js:29]: Since the client will be using some part of the constants to determine if a user is on a landing page, I thought it would be best to keep the constants in the config.

Comment (raisedadead) [reply api-server/src/common/models/user.js:29]: Has the file been moved?

Comment (ahmadabdolsaheb) [reply api-server/src/common/models/user.js:29]: yes

Comment (ahmadabdolsaheb) [reply api-server/src/common/models/user.js:29]: yes

File: cypress/integration/learn/common-components/searchBar.js [hunk 2]
```diff
@@ -27,6 +27,38 @@ describe('Search bar', () => {
     cy.get('.ais-SearchBox').should('be.visible');
   });
 
+  it('Should not display hits', () => {
+    search('freeCodeCamp');
+    cy.get('.ais-Hits-list').should('not.exist');
```
Comment (ojeytonwilliams) [inline cypress/integration/learn/common-components/searchBar.js:32]: [Ideally](https://docs.cypress.io/guides/references/best-practices#Selecting-Elements) we shouldn't rely on implementation details (like classes) for tests.  Could you change this to use a `data-cy` attribute?  E.g. 
```suggestion
    cy.get('.[data-cy=hits-list]').should('not.exist');
```

This applies more generally to the tests in this file, though if those changes would be better in another PR, that's fine.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
I made a few nitpicks, but it looks pretty good.  Shaving 400k of the main bundle is amazing.

File: cypress/integration/learn/common-components/searchBar.js (hunk 2 above)
Comment (ahmadabdolsaheb) [reply cypress/integration/learn/common-components/searchBar.js:32]: Some of the classes that we are targeting are coming straight from the algolia module, but I will add data-cy to what I can.

File: config/constants.js [hunk 3]
```diff
@@ -647,4 +649,5 @@ let blocklist = [
   'zlib'
 ];
 
-export const blocklistedUsernames = [...new Set(blocklist)];
+exports.blocklistedUsernames = [...new Set(blocklist)];
+exports.i18nConstants = i18nConstants;
```
[the remark this answers is not in the document]
Comment (ahmadabdolsaheb) [reply config/constants.js]: Yeah, now that I moved it to config, I started getting 'SyntaxError: Unexpected token 'export'' only on cypress ci 🤷

File: cypress/integration/learn/common-components/searchBar.js (hunk 2 above)
Comment (ojeytonwilliams) [reply cypress/integration/learn/common-components/searchBar.js]: Yeah, I understand.  I think this is slightly safer, though, since no-one is going to add `data-cy` for any reason other than as a target for testing.  Anyways, thanks for implementing this.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Comment by ahmadabdolsaheb** [comment]
@scissorsneedfoodtoo it would be great if you could take a look.

**Review by scissorsneedfoodtoo — APPROVED** [review]
Tested locally and everything LGTM :+1: :+1: Awesome job with these optimizations!

The only difference I noticed between the new search bar and the old one is that the new one opens the search page in a new tab, while the old one redirects there in the same tab. Also that the footer link for the old search bar ("See more results for...") opens the search results page in a new tab.

I think we can iron out that behavior later -- no need hold back your improvements because of my oversight the last time I touched the search bar.

Should we merge this now, or wait for the merge party since there's some TypeScript?

**Comment by ahmadabdolsaheb** [comment]
@scissorsneedfoodtoo, all links that point to an external platfrom should open in a new tab. So, we should make all algolia links to open in a new tab.

**Comment by scissorsneedfoodtoo** [comment]
@ahmadabdolsaheb, awesome, sounds like a plan.
